File: 442/find-all-duplicates-in-an-array.py
# 想找到 array 裡，有沒有重覆的數字
# 因為數字只會出現「一次」或「兩次」所以可用 dict 來做
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        happened = {}
        ans = []
        for num in nums:
            if num in happened:
                ans.append(num)
            happened[num] = True
        return ans

File: 442/find-all-duplicates-in-an-array.c
//想找到重覆的數
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* findDuplicates(int* nums, int numsSize, int* returnSize) {
    int ansN = 0;
    int happened[100001] = {};
    for(int i=0; i<numsSize; i++){
        happened[nums[i]]++;
        if(happened[nums[i]]==2){
            nums[ansN++] = nums[i]; //記錄重覆的數字
        } //回數再利用 nums[]
    }
    *returnSize = ansN;
    return nums;
}
